### PR TITLE
DOC: examples for special functions related to neg. binomial distribution

### PR DESCRIPTION
**Pull request by dschmitz89**
#### Reference issue
#7168 

#### What does this implement/fix?
Examples for:
`special.nbdtr`
`special.nbdtrc`
`special.nbdtri`
`special.nbdtrin`
`special.nbdtrik`

Find the rendered docs [here](https://output.circle-artifacts.com/output/job/d1b78493-3718-4ce8-bc14-6793988341f5/artifacts/0/html/reference/special.html#negative-binomial-distribution)!

### CODE REVIEW

File: scipy/special/_add_newdocs.py [hunk 1]
```diff
@@ -9857,11 +9859,78 @@ def add_newdoc(name, doc):
 
     Wrapper for the Cephes [1]_ routine `nbdtr`.
 
+    The negative binomial distribution is also available as
+    `scipy.stats.nbinom`. Using `nbdtr` directly can compare performance
+    compared to the ``cdf`` method of `scipy.stats.nbinom` (see last example).
```
Comment (j-bowhay) [inline scipy/special/_add_newdocs.py:9864]: Possible typo? Same with the others
```suggestion
    `scipy.stats.nbinom`. Using `nbdtr` directly can improve performance
    compared to the ``cdf`` method of `scipy.stats.nbinom` (see last example).
```

Comment (dschmitz89) [reply scipy/special/_add_newdocs.py]: Good catch, there were even more of those. Fixed in last commit.

**Comment by tupui** [comment]
@j-bowhay Since you started to have a look, are you interested in doing the review?

**Review by j-bowhay — APPROVED** [review]
Had a read through all the examples and checked the rendering; this looks good to go to by my eyes

**Comment by josef-pkt** [comment]
Doesn't look right to me.

negative binomial:
k is random variable
n, p are parameters
e.g. cdf(k ; n, p)

AFAICS:
nbdtrik is the quantile function, inverse of cdf wrt k
npdtri is the inverse of cdf with respect to **parameter** p

e.g. pre-boost code used nbdtrik in ppf method
https://github.com/scipy/scipy/blame/v1.6.0/scipy/stats/_discrete_distns.py#L296

**Comment by dschmitz89** [comment]
@melissawm : friendly ping, I think this is ready to go in :)

**Comment by dschmitz89** [comment]
General question about `nbdtr` and friends: will these cephes based functions at some point removed/deprecated if the distributions anyway use boost?

**Comment by ev-br** [comment]
(a comment from 3000 feet): IMO it would be  nice project to track down duplication and come up with a plan: either deprecate and remove or reimplement old names using new implementations or something. It's frankly a trifle strange to carry three separate erf implementations.

**Comment by dschmitz89** [comment]
I fixed the wrong description found by @josef-pkt .

I guess that while the functions are not used by `stats` anymore they could be replaced by the Boost functions in future but still be available, so better to have the also documented.

**Comment by josef-pkt** [comment]
> IMO it would be nice project to track down duplication and come up with a plan: either deprecate and remove or reimplement old names using new implementations or something. It's frankly a trifle strange to carry three separate erf implementations.

Don't do it as long as boost still needs to fix corner cases.
statsmodels is using some scipy.special functions to avoid problems with boost versions in scipy.stats

Another problem is that the rootfinding functions (for parameters of the distribution) do not roundtrip if they are based on two different numerical approximation. Although that needed only relaxed asserts in statsmodels unit tests.

a bit related: some functions support different dtypes
e.g. norm._cdf in special does not support complex, erf does, AFAIR

I don't know about negbin in special and stats. 
statsmodels does not use it much and I have not seen any problems in what we use.

**Pull request merged**